src/Microsoft.AspNetCore.ResponseCompression/DefaultMimeTypeFilter.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.ResponseCompression
+{
+    /// <inheritdoc/>
+    public class DefaultMimeTypeFilter : IMimeTypeFilter
+    {
+        /// <inheritdoc/>
+        public IMimeTypeFilterConfig Config { get; }
+
+        /// <inheritdoc/>
+        public DefaultMimeTypeFilter(IMimeTypeFilterConfig mimeTypeFilterConfig) =>
+            Config = mimeTypeFilterConfig ?? throw new ArgumentNullException(nameof(mimeTypeFilterConfig));
+
+        /// <inheritdoc/>
+        public virtual bool ShouldCompress(string mimeType) =>
```

Comment:
Assuming the Config gets merged into Options as I suggested elsewhere, this ShouldCompress logic can move into ResponseCompressionProvider. That's already a replaceable component.